### PR TITLE
Adding dynamic config script for ICDS

### PR DESCRIPTION
**Pull request by rohit-dimagi**
##### SUMMARY

* Adding dynamic config script for ICDS

##### ENVIRONMENTS AFFECTED

* ICDS
##### ISSUE TYPE

- Change Pull Request

### CODE REVIEW

File: src/commcare_cloud/ansible/roles/nginx/tasks/install.yml [hunk 1]
```diff
@@ -75,3 +75,36 @@
...
+    hour: "11"
+    user: root
+    cron_file: dynamic_conf_change_nginx
+  when: dynamic_conf_change_nginx is defined
+
+- name: Add Dynamic config change cron reset
+  become: yes
+  cron:
+    name: "Dynamic conf change "
+    job: "/usr/local/bin/dynamic_restore_change_nginx --overall_restore 25 --incremental_restore 25"
```
Comment (snopoke) [inline src/commcare_cloud/ansible/roles/nginx/tasks/install.yml]: can we make these values configurable in the env config?

File: src/commcare_cloud/ansible/roles/nginx/files/dynamic_cfg_change_nginx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    if [[ "$incremental_restore" ]]; then
+            old_incremental_restore=`grep -E '^limit_req_zone.*incremental_restore' $cfg_file`
+            replace_incremental_restore="limit_req_zone \$incremental_sync zone=incremental_restore:10m rate=${incremental_restore}r/s;"
+            sed -i "s[$old_incremental_restore[$replace_incremental_restore[g" $cfg_file
+    fi
+
+    if [[ "$overall_restore" ]]; then
+            old_overall_restore=`grep -E '^limit_req_zone.*overall_restore' $cfg_file`
+            replace_overall_restore="limit_req_zone \$limit_bucket zone=overall_restore:10m rate=${overall_restore}r/s;"
+            sed -i "s[$old_overall_restore[$replace_overall_restore[g" $cfg_file
```
Comment (snopoke) [inline src/commcare_cloud/ansible/roles/nginx/files/dynamic_cfg_change_nginx]: could extract these 3 lines to a function
```
function change_limits() {
    $cfg_file=$1
    $zone=$2
    $var=$3
    $limit=$4
    old_config=`grep -E '^limit_req_zone.*${zone}' ${cfg_file}`
    new_config="limit_req_zone \$${var} zone=${zone}:10m rate=${limit}r/s;"
    sed -i "s[$old_config[$new_config[g" ${cfg_file}
}
change_limits "$cfg_file" "overall_restore" "limit_bucket" "$overall_restore"
```

Comment (rohit-dimagi) [reply src/commcare_cloud/ansible/roles/nginx/files/dynamic_cfg_change_nginx]: Updated

File: src/commcare_cloud/ansible/roles/nginx/tasks/install.yml (hunk 1 above)
Comment (rohit-dimagi) [reply src/commcare_cloud/ansible/roles/nginx/tasks/install.yml]: Made some changes. However i need help on how to iterate over zone. Can you check updated code?

**Comment by rohit-dimagi** [comment]
Updated.

**Pull request merged**